packages/compass-telemetry/src/provider.ts

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { createServiceLocator } from 'hadron-app-registry';
+import type { TrackFunction } from './track';
+
+const noop = () => {
+  // noop
+};
+
+export function createNoopTrack(): TrackFunction {
+  return noop;
+}
+
+const TelemetryContext = React.createContext<{
+  createTrack(): TrackFunction;
+}>({ createTrack: createNoopTrack });
+
+export const TelemetryProvider = TelemetryContext.Provider;
+
+export function createTelemetryLocator() {
+  return createServiceLocator(
+    useTelemetry.bind(null),
+    'createTelemetryLocator'
+  );
+}
+
+export function useTelemetry(): TrackFunction {
+  const context = React.useContext(TelemetryContext);
+  if (!context) {
+    throw new Error('Tracking service is missing from React context');
+  }
+  const trackRef = React.createRef<TrackFunction>();
+  if (!trackRef.current) {
+    (trackRef as any).current = context.createTrack();
+  }
```

Comment:
Why do we need to pass `create` method to the provider instead of just the track method implementation? In logging this is sort of required so that we can pass the `component` value and bond it, but here we're not passing anything

Reply:
we don't, I was actually planning to change that but then I forgot